ToggleItem/ToggleItem.js

```diff
@@ -78,31 +119,135 @@ const ToggleItemBase = kind({
 		css: PropTypes.object,
 
 		/**
-		 * Overrides the icon of the `iconComponent` component.
+		 * Applies a disabled visual state to the toggle item.
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @public
+		 */
+		disabled: PropTypes.bool,
+
+		/**
+		 * An optional prop that lets you override the icon of the `iconComponent` component.
 		 *
-		 * This accepts any string that the [Icon]{@link my-theme/Icon.Icon} component supports,
-		 * provided the recommendations of `iconComponent` are followed.
+		 * This accepts any string that the {@link ui/Icon.Icon|Icon} component supports, provided
+		 * the recommendations of `iconComponent` are followed.
+		 *
+		 * @type {String|Object}
+		 * @public
+		 */
+		icon: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+
+		/**
+		 * Specifies on which side (`'before'` or `'after'`) of `children` the icon appears.
 		 *
-		 * @type {String}
+		 * @type {('before'|'after')}
+		 * @default 'before'
 		 * @public
 		 */
-		icon: PropTypes.string
+		iconPosition: PropTypes.oneOf(['before', 'after']),
+
+		/**
+		 * An additional customizable icon component.
+		 *
+		 * Supports more granular positioning rules. This should only be used *after* specifying the
+		 * `icon` property, as the positioning for this offers the ability to place this in front of
+		 * or behind the existing `icon`. See `itemIconPosition` for options.
+		 *
+		 * @type {Node}
+		 * @public
+		 */
+		itemIcon: PropTypes.node,
+
+		/**
+		 * Specifies where `itemIcon` appears.
+		 *
+		 * * `'before'` - first element in the item
+		 * * `'beforeChildren'` - before `children`. If `iconPosition` is `'before'`, `icon` will be
+		 *	before `itemIcon`
+		 * * `'afterChildren'` - after `children`. If iconPosition` is `'after'`, `icon` will be
+		 *	after `itemIcon`
+		 * * `'after'` - the last element in the item
+		 *
+		 * @type {('before'|'beforeChildren'|'afterChildren'|'after')}
+		 * @default 'afterChildren'
+		 * @public
+		 */
+		itemIconPosition: PropTypes.oneOf(['before', 'beforeChildren', 'afterChildren', 'after']),
```

Comment:
before, itemIconPosition was not a prop and was not sent to ui/ToggleItem. Let's refactor the entire component to not take this into account